File: PaCoSe.Models/Device.cs
using System.Collections.Generic;

namespace PaCoSe.Models
{
    public class Device
    {
        public Device()
        {
            this.DeviceLimits = new List<DeviceLimit>();
        }

        public int Id { get; set; }

        public string IdentifierHash { get; set; }

        public string Name { get; set; }

        public bool IsScreenTimeEnabled { get; set; }

        public string Token { get; set; }

        public List<DeviceLimit> DeviceLimits { get; set; }
    }
}

File: PaCoSe.Web/ClientApp/src/app/common/models/device.ts
import { DeviceLimit } from "./deivce-limits";

export class Device {
    public id: number;
    public name: string;
    public imageUrl: string;
    public token: string;
    public isScreenTimeEnabled: boolean;
    public deviceLimits: DeviceLimit[];
    public typeName: string;
    public typeId: number;
    public isExpanded: boolean;

    constructor(args) {
        this.id = args?.id;
        this.name = args?.name;
        this.imageUrl = args?.imageUrl;
        this.isScreenTimeEnabled = args?.isScreenTimeEnabled || false;
        this.deviceLimits = args && args.deviceLimits?.length ? args.deviceLimits.map(l => new DeviceLimit(l)) : [];
        this.token = args?.token;
        this.typeName = args?.typeName;
        this.typeId = args?.typeId;
    }
}